### PR TITLE
Add notebook: Signature-Aware Model Serving from MLflow with Ray Serve

### PR DESCRIPTION
**Pull request by jinnovation**
# What does this PR do?

This PR adds a notebook demonstrating how to combine two popular members of the OSS MLOps stack -- MLflow and Ray Serve -- in a way that optimizes for developer-friendliness, toil-reduction, and "type-safety" (based on model signatures).

I believe it will be useful to folks creating their own AI platforms based on open-source technologies.

## Who can review?

@merveenoyan; @stevhliu

### CODE REVIEW

File: notebooks/en/mlflow_ray_serve.ipynb [hunk 1]
```diff
@@ -0,0 +1,1079 @@
...
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "I17bSxxg1evl"
+      },
+      "source": [
+        "# Signature-Aware Model Serving from MLflow with Ray Serve\n",
+        "\n",
+        "Authored by: Jonathan Jin"
```
Comment (jinnovation) [inline notebooks/en/mlflow_ray_serve.ipynb:21]:
```suggestion
        "_Authored by: [Jonathan Jin](https://huggingface.co/jinnovation)_"
```

File: notebooks/en/mlflow_ray_serve.ipynb [hunk 2]
```diff
@@ -0,0 +1,1079 @@
+{
```
Comment (jinnovation) [inline notebooks/en/mlflow_ray_serve.ipynb:1]: <p>Let's add the following exercises as well:</p><ul><li>Handle optional values in the MLflow model signature</li><li>Handle <a href="https://mlflow.org/docs/latest/model/signatures.html#model-signatures-with-inference-params" target="_blank">inference parameters</a></li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/cookbook/pull/258/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='43'/>

Comment (stevhliu) [inline notebooks/en/mlflow_ray_serve.ipynb:1]: <p>Nice "motivation" section to really clarify the intent of this notebook</p><p> </p><p>"...can meaningfully drive down..."</p><p> </p><p>"...may not have the benefit of existing infrastructure to form..."</p><p> </p><p>"...deploy a new model, the better."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/cookbook/pull/258/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (stevhliu) [inline notebooks/en/mlflow_ray_serve.ipynb:1]: <p>"...use for our exploration today."</p><p> </p><p>"...but the underlying model architecture..."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/cookbook/pull/258/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

File: notebooks/en/index.md [hunk 3]
```diff
@@ -7,6 +7,7 @@ applications and solving various machine learning tasks using open-source tools
 
 Check out the recently added notebooks:
 
+- [Signature-Aware Model Serving from MLflow with Ray Serve](mlflow_ray_serve)
```
Comment (stevhliu) [inline notebooks/en/index.md:10]: Let's do the following here so only 5 notebooks are shown with the most recent one on top:

```md
- Signature-Aware Model Serving from MLflow with Ray Serve
- Fine-tuning SmolVLM using direct preference optimization (DPO) with TRL on a consumer GPU
- Smol Multimodal RAG: Building with ColSmolVLM and SmolVLM on Colab’s Free-Tier GPU
- Fine-tuning SmolVLM with TRL on a consumer GPU
- Multimodal RAG with ColQwen2, Reranker, and Quantized VLMs on Consumer GPUs
```

File: notebooks/en/_toctree.yml [hunk 4]
```diff
@@ -5,6 +5,12 @@
     - local: index
       title: Overview
 
+    - title: MLOps Recipes
+      isExpanded: false
+      sections:
+        - local: mlflow_ray_serve.ipynb
```
Comment (stevhliu) [inline notebooks/en/_toctree.yml:11]:
```suggestion
        - local: mlflow_ray_serve
```

**Review by stevhliu — COMMENTED** [review]
Really cool to see a more production/deployment type notebook, great job!

File: notebooks/en/mlflow_ray_serve.ipynb (hunk 2 above)
Comment (jinnovation) [reply notebooks/en/mlflow_ray_serve.ipynb:1]: <p>Done!</p>

File: notebooks/en/index.md (hunk 3 above)
Comment (jinnovation) [reply notebooks/en/index.md:10]: Done

File: notebooks/en/mlflow_ray_serve.ipynb (hunk 2 above)
Comment (jinnovation) [reply notebooks/en/mlflow_ray_serve.ipynb:1]: Done!

**Comment by jinnovation** [comment]
> Really cool to see a more production/deployment type notebook, great job!

Glad you like it! Do you have a preference between calling this new section "MLOps Recipes" or "Production ML Recipes"? I'm torn between the two.

**Comment by stevhliu** [comment]
I like MLOps more because it feels like it is broader and more inclusive of other topics in this space 🙂

**Comment by jinnovation** [comment]
@stevhliu: Sounds good, will leave it as is then.

Any further changes you'd like to see before we merge this in?

**Review by stevhliu — APPROVED** [review]
LGTM, thanks for your contribution! 🫡

**Pull request merged**